### PR TITLE
Updated AGW-4 ARG

### PR DESCRIPTION
**Pull request by taranki**
# Overview/Summary

Updated AppGW main page to indicate AGW-4 has an ARG query, and adds the AGW-4 query for determining AGW deployments not on v2

## Related Issues/Work Items

AB#34283
(https://dev.azure.com/CSUSolEng/Azure%20Infra%20Networking/_workitems/edit/34283/)

## This PR fixes/adds/changes/removes

1. Updates main AGW page to indicate ARG query for AGW-4
2. Updates ARG query for AGW-4

### Breaking Changes
None

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by taranki** [comment]
Screenshot of the query running, but I am unable to deploy a v1 AGW resource as they are deprecated and no longer deployable unless you have current v1 deployments in your subscription. See [Https://aka.ms/v1](https://aka.ms/V1retirement)

Screenshot 2024-03-19 130422

Updating the query to look for a different version (for example v1) shows results indicating that the current query is excluding v2 deployed resources only.
Screenshot 2024-03-19 130500

File: docs/content/services/networking/application-gateway/code/agw-4/agw-4.kql [hunk 1]
```diff
@@ -1 +1,6 @@
-// under-development
+// Azure Resource Graph Query
+// This query will return all Application Gateways not on V2
+resources
+| where type =~ 'microsoft.network/applicationgateways'
+| where properties.sku.name !contains 'v2'
+| project recommendationId = "agw-4", name, id, tags, param1="Application Gateway v2 should be used unless there is a compelling reason for using v1."
```
Comment (ejhenry) [inline docs/content/services/networking/application-gateway/code/agw-4/agw-4.kql:6]: Generally, we use the param columns to provide additional detail on the resource to indicate why we are highlighting it for the recommendation. With this recommendation it's probably not necessary and we could remove param1 entirely from the query.

**Comment by taranki** [comment]
Removed param1 from the query as advised

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Pull request merged**